controllers/factory/k8stools/placeholders.go

```diff
@@ -0,0 +1,34 @@
+package k8stools
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+func RenderPlaceholders[T any](resource *T, placeholders map[string]string) (*T, error) {
+	data, err := json.Marshal(resource)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal resource for filling placeholders: %w", err)
+	}
+
+	if len(placeholders) == 0 || len(data) == 0 {
```

Comment:
It could be moved into fast path, e.g.:

func Render...(arg [])
 if len(arg) == {
     return
  }
 In this case, we doesn't call Marshal and save some cpu and memory.

 Think, that check len(data) == 0 is not needed or we can check resource for nil at fast path

Reply:
fixed